pom.xml

```diff
@@ -88,6 +88,13 @@
       <version>${lib.assertj.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.immutables</groupId>
+      <artifactId>encode</artifactId>
+      <version>${lib.immutables.version}</version>
+      <scope>provided</scope>
+      <!-- compile only -->
```

Comment:
Why is this labeled as compile only? When scope is thinking it's provided. Kind of confusing.

Reply:
Because maven doesn't have a compile time only scope. https://immutables.github.io/getstarted.html

So, we comment this because we're just using `provided` as a workaround for the lack of that.

Reply:
https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope
Is the provided `compile` insufficient for what we need?

Reply:
"those dependencies are propagated to dependent projects." is what we're avoiding by using provided.

Reply:
Alright gotcha.